lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -788,6 +788,18 @@ mlir::LogicalResult mlir::triton::ReshapeOp::verify() {
   return mlir::success();
 }
 
+//-- FpToFpOp --
+mlir::LogicalResult mlir::triton::FpToFpOp::verify() {
+  auto dstType = getType().cast<RankedTensorType>().getElementType();
```

Comment:
from the definition of the op the source and destination can be a tensor or a scalar, that would crash for the scalar case.

Reply:
Good catch!

Reply:
Actually, as discussed offline, FpToFpOp cannot be currently generated for non-tensor types, I have changed the Op definition to reflect that and be consistent with the rest of the code.

Reply:
Nice :)